test/Microsoft.AspNet.Mvc.Core.Test/SessionStateTempDataProviderTest.cs

```diff
@@ -201,20 +189,150 @@ public void SaveAndLoad_SimpleTypesCanBeStoredAndLoaded()
             // Assert
             var stringVal = Assert.IsType<string>(TempData["string"]);
             Assert.Equal("value", stringVal);
+        }
+
+        [Fact]
+        public void SaveAndLoad_IntCanBeStoredAndLoaded()
+        {
+            // Arrange
+            var testProvider = new SessionStateTempDataProvider();
+            var input = new Dictionary<string, object>
+            {
+                { "int", 10 }
+            };
+            var context = GetHttpContext(new TestSessionCollection(), true);
+
+            // Act
+            testProvider.SaveTempData(context, input);
+            var TempData = testProvider.LoadTempData(context);
+
+            // Assert
             var intVal = Convert.ToInt32(TempData["int"]);
             Assert.Equal(10, intVal);
+        }
+
+        [Fact]
+        public void SaveAndLoad_BoolCanBeStoredAndLoaded()
```

Comment:
Can we make this a `Theory` so we know both `true` and `false` work (and it's not returning `default(bool)`?